backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/model/validator/M_HU_PI.java

```diff
@@ -47,4 +46,22 @@ public void onDeleteMHUPI(final I_M_HU_PI pi)
 			InterfaceWrapperHelper.delete(version);
 		}
 	}
+
+	@ModelChange(timings = { ModelValidator.TYPE_BEFORE_NEW, ModelValidator.TYPE_BEFORE_CHANGE },
+			ifColumnsChanged = I_M_HU_PI.COLUMNNAME_IsDefaultForPicking
+	)
+	public void ensureOnlyOneDefaultForPicking(@NonNull final I_M_HU_PI newDefault)
+	{
+		if (!newDefault.isDefaultForPicking())
+		{
+			return;
+		}
+
+		final I_M_HU_PI previousDefault = Services.get(IHandlingUnitsDAO.class).retrievePIDefaultForPicking();
+		if (previousDefault != null)
+		{
+			previousDefault.setIsDefaultForPicking(false);
+			InterfaceWrapperHelper.save(previousDefault);
+		}
```

Comment:
```suggestion
		try (final MDCCloseable ignored = TableRecordMDC.putTableRecordReference(newDefault))
		{
			if (!newDefault.isDefaultForPicking())
			{
				return;
			}

			final I_M_HU_PI previousDefault = Services.get(IHandlingUnitsDAO.class).retrievePIDefaultForPicking();
			if (previousDefault != null)
			{
				logger.debug("M_HU_PI={} is now IsDefaultForPicking; -> Change previousDefault M_HU_PI={} to IsDefaultForPicking='N'",
						newDefault.getM_HU_PI_ID(), previousDefault.getM_HU_PI_ID());
				previousDefault.setIsDefaultForPicking(false);
				InterfaceWrapperHelper.save(previousDefault);
			}
		}
```

Reply:
also, instead of 

> InterfaceWrapperHelper.save(previousDefault);

pls introduce and use: IHandlingUnitsDAO.save(I_M_HU_PI)

Reply:
done and done